pkg/clients/ecr/repository_policy.go

```diff
@@ -146,6 +148,15 @@ func SerializeRepositoryPrincipal(p *v1alpha1.RepositoryPrincipal) (interface{},
 func SerializeAWSPrincipal(p v1alpha1.AWSPrincipal) *string {
 	switch {
 	case p.AWSAccountID != nil:
+		// Note: AWS Docs say you can specify the account ID either
+		// raw or as an ARN, but AWS actually converts internally to
+		// the ARN format, which is problematic for checking if we're
+		// up to date. So here we just do the conversion ourselves to
+		// avoid unintended behaviour if user specifies raw account ID.
+		if !strings.HasPrefix(*p.AWSAccountID, "arn:aws:iam::") {
+			s := fmt.Sprintf("arn:aws:iam::%s:root", *p.AWSAccountID)
```

Comment:
Is this always `root`?

Reply:
I think when specifying only the account ID, yes - `root` is assumed (from the IAM docs) and means any role in the account in question:

```
For example, given an account ID of 123456789012, you can use either of the following methods to specify that account in the Principal element:

"Principal": { "AWS": "arn:aws:iam::123456789012:root" }

"Principal": { "AWS": "123456789012" }
```

Reply:
I think it _might_ be a good idea to check if the account id is numeric only before doing this re-format. It will be more future-compatible and if someone makes mistake, like copy the output from "get-caller-identity", it will not format it, for example `arn:aws:sts::123:root` as `arn:aws:iam::arn:aws:sts:123:root:root`

Reply:
@chlunde Yeah that's a fair point. Given that AWS account numbers are 12 numeric digits I think we can just do the following:

```go
if _, err := strconv.ParseInt(*p.AWSAccountID, 10, 64); err == nil {
	s := fmt.Sprintf("arn:aws:iam::%s:root", *p.AWSAccountID)
	return &s
}
```

rather than checking for a string prefix. Anything with non-numerics should error and fall back to no re-format.